pkg/apis/core/v1beta1/helper/helper.go

```diff
@@ -722,6 +727,22 @@ func ReadShootedSeed(shoot *gardencorev1beta1.Shoot) (*ShootedSeed, error) {
 	return shootedSeed, nil
 }
 
+// GetManagedSeed gets the ManagedSeed resource for this Shoot. If Spec.ManagedSeedName is nil or a ManagedSeed resource
+// with the specified name is not found, nil is returned.
+func GetManagedSeed(ctx context.Context, c client.Client, shoot *gardencorev1beta1.Shoot) (*seedmanagementv1alpha1.ManagedSeed, error) {
+	if shoot.Spec.ManagedSeedName == nil {
+		return nil, nil
+	}
+	managedSeed := &seedmanagementv1alpha1.ManagedSeed{}
+	if err := c.Get(ctx, kutil.Key(shoot.Namespace, *shoot.Spec.ManagedSeedName), managedSeed); err != nil {
+		if apierrors.IsNotFound(err) {
```

Comment:
My thinking was that if the `managedSeedName` points to a resource that doesn't exist, then the shoot is not really registered as a seed and could be reconciled as an "ordinary" shoot, instead of throwing an error. But I guess it's indeed better to fail in this case, so I will change it.